crates/bevy_reflect/src/array.rs

```diff
@@ -358,6 +352,46 @@ impl Array for DynamicArray {
     }
 }
 
+impl FromIterator<Box<dyn Reflect>> for DynamicArray {
+    fn from_iter<I: IntoIterator<Item = Box<dyn Reflect>>>(values: I) -> Self {
+        Self {
+            represented_type: None,
+            values: values.into_iter().collect::<Vec<_>>().into_boxed_slice(),
+        }
+    }
+}
+
+impl<T: Reflect> FromIterator<T> for DynamicArray {
+    fn from_iter<I: IntoIterator<Item = T>>(values: I) -> Self {
```

Comment:
I guess this is mostly a matter of preference, but these could map the iterator and then defer to the `Box<dyn Reflect>` version if you wanted; e.g.:
```rust
        Self::from_iter(
            values
                .into_iter()
                .map(|field| Box::new(field).into_reflect()),
        )
```